corva/state/redis_adapter.py

```diff
@@ -14,64 +11,52 @@ class RedisAdapter(Redis):
 
     def __init__(
          self,
-         default_name: str,
+         name: str,
          cache_url: str,
-         logger: Union[Logger, LoggerAdapter] = DEFAULT_LOGGER,
          **kwargs
     ):
         kwargs.setdefault('decode_responses', True)
         super().__init__(connection_pool=from_url(url=cache_url, **kwargs).connection_pool)
-        self.logger = logger
-        self.default_name = default_name
+        self.name = name
         try:
             self.ping()
         except ConnectionError as exc:
             raise ConnectionError(f'Could not connect to Redis with URL: {cache_url}') from exc
 
     def hset(
          self,
-         name: Optional[str] = None,
          key: Optional[str] = None,
          value: Optional[REDIS_STORED_VALUE_TYPE] = None,
          mapping: Optional[Dict[str, REDIS_STORED_VALUE_TYPE]] = None,
          expiry: Union[int, timedelta, None] = DEFAULT_EXPIRY
     ) -> int:
-        name = name or self.default_name
-
-        n_set = super().hset(name=name, key=key, value=value, mapping=mapping)
+        n_set = super().hset(name=self.name, key=key, value=value, mapping=mapping)
 
-        if expiry is None and self.pttl(name=name) > 0:
-            self.persist(name=name)
+        if expiry is None and self.pttl() > 0:
+            self.persist(name=self.name)
 
         if expiry is not None:
-            self.expire(name=name, time=expiry)
+            self.expire(name=self.name, time=expiry)
 
         return n_set
 
-    def hget(self, key: str, name: Optional[str] = None) -> Union[REDIS_STORED_VALUE_TYPE, None]:
-        name = name or self.default_name
-        return super().hget(name=name, key=key)
+    def hget(self, key: str) -> Union[REDIS_STORED_VALUE_TYPE, None]:
+        return super().hget(name=self.name, key=key)
 
-    def hgetall(self, name: Optional[str] = None) -> Dict[str, Union[REDIS_STORED_VALUE_TYPE]]:
-        name = name or self.default_name
-        return super().hgetall(name=name)
+    def hgetall(self) -> Dict[str, Union[REDIS_STORED_VALUE_TYPE]]:
+        return super().hgetall(name=self.name)
 
-    def hdel(self, keys: List[str], name: Optional[str] = None) -> int:
-        name = name or self.default_name
-        return super().hdel(name, *keys)
+    def hdel(self, keys: List[str]) -> int:
+        return super().hdel(self.name, *keys)
 
-    def delete(self, *names: List[str]) -> int:
-        names = names or [self.default_name]
-        return super().delete(*names)
+    def delete(self) -> int:
+        return super().delete(self.name)
```

Comment:
Is there any way we can use a non-modified Redis adapter?

Reply:
yes, we can. logic from overridden functions in RedisAdapter can be moved to corresponding functions in RedisState (e.g move logic from RedisAdapter.hset to RedisState.store). we will get rid of RedisAdapter class and logic will become more straightforward.